src/keystore/InMemoryKeystore.ts

```diff
@@ -251,28 +254,139 @@ export default class InMemoryKeystore implements Keystore {
-      const sealed = await SealedInvitation.createV1({
-        sender: this.v2Keys,
+
+      return await this.makeInvite(
+        this.v2Keys,
         recipient,
         created,
-        invitation,
-      })
-      const topicData = {
-        invitation,
-        createdNs: req.createdNs,
-        peerAddress: await recipient.walletSignatureAddress(),
+        req.createdNs,
+        invitation
+      )
+    } catch (e) {
+      throw convertError(e as Error, ErrorCode.ERROR_CODE_INVALID_INPUT)
+    }
+  }
+
+  async createInvites(
+    req: keystore.CreateInvitesRequest
+  ): Promise<keystore.CreateInviteResponse[]> {
+    try {
+      if (!validateObject(req, ['recipients'], [])) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_INVALID_INPUT,
+          'missing recipients'
+        )
       }
-      await this.inviteStore.add([topicData])
 
-      return keystore.CreateInviteResponse.fromPartial({
-        conversation: topicDataToConversationReference(topicData),
-        payload: sealed.toBytes(),
+      // Register an address on the network for the group
+      const sharedWallet = Wallet.createRandom()
+      const sharedClient = await Client.create(sharedWallet)
+      const sharedPrivateKeyBundle =
+        await sharedClient.keystore.getPrivateKeyBundle()
+
+      if (!sharedPrivateKeyBundle) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_INVALID_INPUT,
+          'missing private key bundle'
+        )
+      }
+
+      const invitation = InvitationV1.createRandom(req.context)
+      const created = nsToDate(req.createdNs)
+      const recipients = req.recipients.map(toSignedPublicKeyBundle)
+
+      return Promise.all(
+        recipients.map(async (recipient) => {
+          return await this.makeInvite(
+            PrivateKeyBundleV2.fromLegacyBundle(
+              new PrivateKeyBundleV1(sharedPrivateKeyBundle)
+            ),
+            recipient,
+            created,
+            req.createdNs,
+            invitation
+          )
+        })
+      )
+    } catch (e) {
+      throw convertError(e as Error, ErrorCode.ERROR_CODE_INVALID_INPUT)
+    }
+  }
+
+  async createInviteFromTopic(
+    req: keystore.CreateInviteFromTopicRequest
+  ): Promise<keystore.CreateInviteResponse> {
+    try {
+      if (!validateObject(req, ['contentTopic'], [])) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_INVALID_INPUT,
+          'missing topic'
+        )
+      }
+
+      const topicData = this.inviteStore.lookup(req.contentTopic)
+      if (!topicData) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_INVALID_INPUT,
+          'missing topic data'
+        )
+      }
+
+      const invitation = new InvitationV1({
+        context: topicData.invitation.context,
+        topic: req.contentTopic,
+        aes256GcmHkdfSha256: topicData.invitation.aes256GcmHkdfSha256,
       })
+
+      const created = nsToDate(req.createdNs)
```

Comment:
One interesting quirk here is that the invitation timestamp and the envelope's timestamp need to match or `SealedInvitation.fromEnvelope` will fail